src/routes/intro/__tests__/ProfessionSelectionScreen.spec.tsx

```diff
@@ -0,0 +1,73 @@
+import { RouteProp } from '@react-navigation/native'
+import { fireEvent } from '@testing-library/react-native'
+import { mocked } from 'jest-mock'
+import React from 'react'
+
+import labels from '../../../constants/labels.json'
+import { useLoadDisciplines } from '../../../hooks/useLoadDisciplines'
+import useReadSelectedProfessions from '../../../hooks/useReadSelectedProfessions'
+import { RoutesParams } from '../../../navigation/NavigationTypes'
+import AsyncStorageService from '../../../services/AsyncStorage'
+import createNavigationMock from '../../../testing/createNavigationPropMock'
+import { getReturnOf } from '../../../testing/helper'
+import { mockDisciplines } from '../../../testing/mockDiscipline'
+import render from '../../../testing/render'
+import ProfessionSelectionScreen from '../ProfessionSelectionScreen'
+
+jest.mock('@react-navigation/native')
+jest.mock('../../../hooks/useLoadDisciplines')
+jest.mock('../../../hooks/useReadSelectedProfessions')
+
+describe('ProfessionSelectionScreen', () => {
+  const navigation = createNavigationMock<'ProfessionSelection'>()
+  const route: RouteProp<RoutesParams, 'ProfessionSelection'> = {
+    key: 'key-1',
+    name: 'ProfessionSelection',
+    params: {
+      discipline: {
+        id: 5,
+        title: 'Parent Discipline',
+        description: 'Parent Description',
+        icon: 'none',
+        numberOfChildren: 2,
+        isLeaf: false,
+        parentTitle: null,
+        needsTrainingSetEndpoint: false
+      }
+    }
+  }
+
+  it('should select profession when clicked', async () => {
+    await AsyncStorageService.setSelectedProfessions([])
+    mocked(useLoadDisciplines).mockReturnValueOnce(getReturnOf(mockDisciplines))
+    mocked(useReadSelectedProfessions).mockReturnValueOnce(getReturnOf(null))
+    const spyOnPush = jest.spyOn(AsyncStorageService, 'pushSelectedProfession')
+
+    const { findByText, queryAllByTestId } = render(<ProfessionSelectionScreen route={route} navigation={navigation} />)
+    expect(await findByText(labels.intro.skipSelection)).toBeDefined()
+    const profession = await findByText(mockDisciplines[0].title)
+    expect(profession).toBeDefined()
+    expect(queryAllByTestId('check-icon')).toHaveLength(0)
+    fireEvent.press(profession)
+    expect(spyOnPush).toHaveBeenCalledTimes(1)
+    expect(spyOnPush).toHaveBeenCalledWith(mockDisciplines[0])
```

Comment:
Optional to avoid spies:
```suggestion
    await waitFor(async () => {
      const selectedProfessions = await AsyncStorageService.getSelectedProfessions()
      expect(selectedProfessions).toEqual([mockDisciplines[0]])
    })

```

Reply:
Done.
I decided to use a spy, as I thought it makes sense for a unit test, to just check if "outgoing" call is valid and not also if the AsyncStorageService is working properly. Why is it bad to use spies?
Best test case, would be to wait for green-check-icon to appear / disapear, but I cannot get this working.

Reply:
I don't know if it is bad to use spies but I think it kind of tests implementation details which could be avoided here checking if the async storage has the right value afterwards. E.g. it could happen that while the spy is called correctly, another async storage function is called somewhere else overwriting it in the async storage.